airflow/migrations/versions/0037_3_0_0_add_sdm_foreignkey_to_dagrun.py

```diff
@@ -0,0 +1,80 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+Add SDM foreign key to DagRun, TI & TIH.
```

Comment:
Just wondering. We add SDM foreign key to DagRun, is it necessary to also add it to TI ?

Can the SDM change in between tasks and is that the reason to do so ? Naively I would think that the SDM is the same for the DagRun and it's TIs ?

Reply:
Yeah, @pierrejeambrun , it's possible to change in between tasks. The scheduler constantly checks for changes in the SDM and updates the dag_run with the new SDM. Why it was added to TI is because when we clear, we might be running with a different SDM and recording the previous one will be necessary, hence the TIH. 
By the way, I'm exploring an alternative to this, which would have the DagVersion as a separate table

Reply:
Got it, thanks for the details :)